src/main/java/name/webdizz/sonar/grammar/spellcheck/JavaSourceCodeWordFinder.java

```diff
@@ -3,16 +3,22 @@
 import com.swabunga.spell.event.AbstractWordFinder;
 import com.swabunga.spell.event.Word;
 import com.swabunga.spell.event.WordNotFoundException;
+import name.webdizz.sonar.grammar.PluginParameter;
+import org.sonar.api.BatchExtension;
+import org.sonar.api.config.Settings;
 
-public class JavaSourceCodeWordFinder extends AbstractWordFinder {
+public class JavaSourceCodeWordFinder extends AbstractWordFinder implements BatchExtension {
     private boolean inComment;
+    private int minimumWordLength;
+    private Settings settings;
 
     public JavaSourceCodeWordFinder() {
         super();
     }
 
-    public JavaSourceCodeWordFinder(final String searchText) {
+    public JavaSourceCodeWordFinder(final String searchText, final Settings settings) {

```

Comment:
It seems here is the naswer http://picocontainer.codehaus.org/faq.html
As described - during bean creation pico uses the most "greediest" constructor. Why it is in no invoked, I'm not sure, probably we can't debug this moment? But it is invoked because if it isn't inwoked we would never get minimumWordLength set, and our tests would not be passed, but they are.